src/main/java/com/squareup/javapoet/ParameterizedTypeName.java

```diff
@@ -49,7 +49,8 @@
   }
 
   @Override public ParameterizedTypeName annotated(List<AnnotationSpec> annotations) {
-    return new ParameterizedTypeName(rawType, typeArguments, concatAnnotations(annotations));
+    ClassName annotatedRawType = rawType.annotated(annotations);
```

Comment:
Here is the main difference in regard to #436 - the underlying raw type is annotated. This might be somehow ... unexpected. But simplifies the internal annotation emission, when compared to the solution proposed in #436